tests/pact/caps.repl

```diff
@@ -41,6 +45,14 @@
   (defun enforce-msg-keyset (key:object{guardschema})
     (enforce-keyset (read-keyset (at 'key key))))
 
+  ; a user guard which reads from the db, which we want to disallow.
+  (defun db-user-guard ()
+    (insert ints 'x {'i: 0})
```

Comment:
This `insert` is fine here per the test on line 162. The guard is first created in an unrestricted context (where this is allowed), and _then_ `enforceGuard` occurs in a pure context.

Reply:
Naming makes this hard to understand. I suggest:
- rename this to `create-bad-db-user-guard`, and give the context "insert should succeed" in the code comment (also, put that in the docstring instead of as a LISP comment above the defun)
- rename `read-zero-from-db` to `bad-user-guard-fun` or something.
Ideally the test below would let-bind the guard to ensure that the create function succeeds before failing with the enforce. A weakness of `expect-failure` is that things can fail for the wrong reason. We should consider adding error string matching (not full matching b/c that's brittle).